crates/rustc_codegen_spirv/src/builder/byte_addressable_buffer.rs

```diff
@@ -43,7 +42,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 u32_ptr,
                 None,
                 array.def(self),
-                IntoIter::new([actual_index.def(self)]),
+                IntoIterator::into_iter([actual_index.def(self)]),
```

Comment:
`in_bounds_access_chain` takes `impl IntoIterator`, so I think this can just pass the array now.